File: loadercomplex-forge-18/src/main/java/com/enderzombi102/loadercomplex/forge18/impl/block/ForgeBlockstate.java
package com.enderzombi102.loadercomplex.forge18.impl.block;

import com.enderzombi102.loadercomplex.api.block.Blockstate;
import com.enderzombi102.loadercomplex.api.utils.ResourceIdentifier;
import net.minecraft.block.BlockState;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import net.minecraftforge.registries.RegistryManager;

public class ForgeBlockstate implements Blockstate {
	private final BlockState backingState;

	public ForgeBlockstate(BlockState state) {
		backingState = state;
	}

	@Override
	public ResourceIdentifier getBlockType() {
		Identifier id = RegistryManager.ACTIVE.getRegistry( Registry.BLOCK_KEY ).getKey( backingState.getBlock() );
		return new ResourceIdentifier( id.getNamespace(), id.getPath() );
	}

	@Override
	public Object getObject() {
		return backingState;
	}
}

File: loadercomplex-forge/src/ver182/kotlin/com/enderzombi102/loadercomplex/forge182/impl/block/ForgeBlockState.kt
package com.enderzombi102.loadercomplex.forge182.impl.block

import com.enderzombi102.loadercomplex.api.minecraft.util.ResourceIdentifier
import net.minecraft.block.BlockState
import net.minecraft.util.registry.Registry
import net.minecraftforge.registries.RegistryManager

class ForgeBlockState(private val backingState: BlockState) :
	com.enderzombi102.loadercomplex.api.minecraft.block.BlockState {
	override fun getBlockType(): ResourceIdentifier {
		val id = RegistryManager.ACTIVE.getRegistry(Registry.BLOCK_KEY).getKey(backingState.block)
		return ResourceIdentifier(id!!.namespace, id.path)
	}

	override fun getObject(): Any {
		return backingState
	}
}
